### ISSUE TITLE
Want to overwrite an array of document

### ISSUE THREAD

**Issue by akanshgulati** [Automattic/mongoose#4492]
I want to overwrite an array of document having duplicate key used for indexing. 
Schema is 
````
    "id": {type : Number, required: true, unique: true, dropDups: true},

````
I am using static functions for it
````
agentMethods.statics.createAgents = function (data, cb) {
    return this.create(data, cb);
};
````
The `data` can have some new objects and some old objects while creating a array in DB.

Is there any field I can specify in DB which can overwrite the existing document? Since data will be like 30-40 documents, so I want it to be fast too.

**Comment by vkarpov15** [comment]
Can you please clarify? I don't quite understand your question

**Comment by akanshgulati** [comment]
@vkarpov15 I have been storing an array of document and some of the elements in that array are already present in MongoDB. Now I want the old one to be replaced and the new one to be inserted just by one command. 
I came across `update` query with `upsert` option but not sure if that's the perfect way of achieving that. After some search, I came to know that `DropDups` is deprecated in newer versions. 
Just let me know if there is any field or better wayout for achieving the same for saving array of documents.
